### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by masterleinad**
We need a few more packages for #577. Also use Trilinos 13 for testing.

### CODE REVIEW

**Comment by aprokop** [comment]
Why is this a separate PR? Why not merge it as part of #577?

**Comment by masterleinad** [comment]
If we want to discuss about using the Dockerfile or updating Trilinos, I would rather do it here to keep things separate. I don't feel strongly either way.

File: docker/Dockerfile [hunk 1]
```diff
@@ -170,8 +170,8 @@ ENV NETCDF_DIR=/opt/netcdf/4.7.3
 
 # download Trilinos
 # Current hash has Kokkos 3.1
-RUN export TRILINOS_HASH=7b530eda6fdda0a4f17d192dfe7c4e07d582110e && \
-    export TRILINOS_SHORT_HASH=7b530eda && \
+RUN export TRILINOS_HASH=9fec35276d846a667bc668ff4cbdfd8be0dfea08 && \
```
Comment (aprokop) [inline docker/Dockerfile:173]: Add a comment that it corresponds to 13.0.0 release

File: docker/Dockerfile [hunk 2]
```diff
@@ -218,7 +221,7 @@ RUN export TRILINOS_HASH=7b530eda6fdda0a4f17d192dfe7c4e07d582110e && \
       -D Kokkos_ENABLE_Cuda_Lambda=ON \
       -D Kokkos_ARCH_VOLTA70=ON \
       -D Tpetra_INST_CUDA=ON \
-      -D Trilinos_CXX11_FLAGS="-std=c++11 --expt-extended-lambda" \
+      -D Trilinos_CXX11_FLAGS="-std=c++14 --expt-extended-lambda" \
```
Comment (aprokop) [inline docker/Dockerfile:224]: Please explain why this is needed in this PR.

Comment (masterleinad) [reply docker/Dockerfile:224]: Ah, good point. This should not have any effect any more, see https://github.com/trilinos/Trilinos/pull/6814.

Comment (aprokop) [reply docker/Dockerfile]: While this is a helpful reference, the main thrust of my question was why this PR changes the standard from 11 to 14 without a need? As it does not update ArborX, it seems that such a such is unwarranted here if it worked with 11 before.

Comment (masterleinad) [reply docker/Dockerfile]: My thinking is that if we want C++14 anyway it's a good idea to also tell Kokkos and whatever else in Trilinos would do something else for C++14. I don't want to fight over this at all, though. If you prefer I can just drop specifying any standard at all or even avoid updating Trilinos.

Comment (aprokop) [reply docker/Dockerfile]: I'm not opposed to it, just wanted to understand if this change is required or not.

File: docker/Dockerfile [hunk 3]
```diff
@@ -168,10 +168,10 @@ RUN export NETCDF_VERSION=4.7.3 && \
 
 ENV NETCDF_DIR=/opt/netcdf/4.7.3
 
-# download Trilinos
+# download Trilinos 13
```
Comment (aprokop) [inline docker/Dockerfile:171]:
```suggestion
# download Trilinos 13.0.0
```

**Review by aprokop — APPROVED** [review]

**Comment by aprokop** [comment]
Yeah, OK, maybe switching to building Docker image wasn't the brightest idea. With additional Trilinos packages enabled it takes close to an hour.

**Pull request merged**